### ISSUE TITLE
Logs are not written in Query Log File in Thanos

### ISSUE THREAD

**Issue by SanthoshiMekala** [thanos-io/thanos#5901]
Hi Team,

We have recently migrated from traditional Prometheus to Thanos. In our Thanos configuration, we have multiple replicas of prometheus-shard. We configured query_log_file for our traditional Prometheus like below and it was working fine. We used the similar kind of configuratin for Prometheus-shard also, but no logs are being writted in the query_log_file. Please let me know if I am missing anything here.
global:
      query_log_file: /prometheus/query.log

Thanos Version: v0.24.0-b1.0
Prometheus Version: v2.34.0-b4.0

Thanks in advance!

**Comment by hanjm** [comment]
thanos use remote_read api to interact with prometheus, not the normal query api, prometheus query_log_file only log the request via query api.
you can use thanos query log or tracing to record all the request.
doc:
logging:  https://thanos.io/tip/thanos/logging.md/
tracing: https://thanos.io/tip/thanos/tracing.md/

**Comment by saswatamcode** [comment]
Yes, you can also use active query tracking to see the queries that are currently being processed by the Thanos Querier: https://thanos.io/tip/components/query.md/#active-query-tracking

**Comment by matej-g** [comment]
Looks this has been answered, closing for now!

**Comment by SanthoshiMekala** [comment]
Hi All,
I tried using thanos query logging like mentioned in the doc (https://thanos.io/tip/thanos/logging.md/), I checked the files inside the tsdb.path. But the logs are not being written. Could you please let me know where to check the logs after enabling thanos logging?
And I tried with the  active query tracking, but I see the queries.active file looks like encrypted and not readable. Could you please let me know if I need to do any more configuration.
Thanks in advance!

**Comment by hanjm** [comment]
the logging is write to stdout/stderr, not files in tsdb.path.

**Comment by SanthoshiMekala** [comment]
Hi Jimmie,

I checked the /dev/null in both prometheus and thanos-sidecar containers, I don't see the logs written there. Please have a look at my configuration for thanos-sidecar in prometheus-shard pod. Please let me know if I am missing anything here.
- args:
        - sidecar
        - --log.level=debug
        - |
          --request.logging-config=http:
            config:
              - path: /api/v1/query
                port: 10901
            options:
              level: DEBUG
              decision:
                log_start: true
                log_end: true
        - --tsdb.path=/prometheus
        - --prometheus.url=http://127.0.0.1:1234/prometheus

**Comment by saswatamcode** [comment]
@SanthoshiMekala there is no separate file for logs, it is just printed to stdout/stderr. You can get/stream logs from your Thanos pods using kubectl: `kubectl logs -f <name_of_pod>`

**Comment by SanthoshiMekala** [comment]
Hi @saswatamcode I see the normal logs. But I don't see any logs related to the queries. In old query.log file, we used to have the data in json format and we can get data like how much time each query is taking etc... We want that kind of information. Could you please provide if there is any way like that?

**Comment by SanthoshiMekala** [comment]
And could anyone please let me know how to read the queries.active file? Currently its not in readable format.

**Comment by yeya24** [comment]
@SanthoshiMekala From the code looks like json entries are logged into the file. I haven't really tried reading the active tracker though and couldn't find any documentation about that in Prometheus.
